include/boost/geometry/algorithms/detail/num_distinct_consecutive_points.hpp

```diff
@@ -48,22 +48,20 @@ struct num_distinct_consecutive_points
     template <typename Strategy>
     static inline std::size_t apply(Range const& range, Strategy const& strategy)
     {
-        typedef typename boost::range_iterator<Range const>::type iterator;
-
         std::size_t const size = boost::size(range);
 
         if ( size < 2u )
         {
             return (size < MaximumNumber) ? size : MaximumNumber;
         }
 
-        iterator current = boost::begin(range);
-        iterator const end = boost::end(range);
+        auto current = boost::begin(range);
+        auto const end = boost::end(range);
         std::size_t counter(0);
         do
         {
             ++counter;
-            iterator next = std::find_if(current, end, [&](auto const& pt) {
+            auto next = std::find_if(current, end, [&](auto const& pt) {
```

Comment:
Though here the names doesn't bother me because I can see what they are from the context.